File: mergeTwoSortedLists/mergeTwoSortedLists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */


public ListNode MergeTwoLists(ListNode l1, ListNode l2) {
    ListNode head = new ListNode();
    ListNode current = head;
    
    while (l1 != null && l2 != null) 
    {
        if (l1.val > l2.val)
        {
            current.next = new ListNode(l2.val);                
            l2 = l2.next;
        }
        else 
        {
            current.next = new ListNode(l1.val);                
            l1 = l1.next;
        }   
        current = current.next;
    }
    while (l1 != null)
    {
        current.next = new ListNode(l1.val);
        current = current.next;
        l1 = l1.next;
    }
        
    while (l2 != null)
    {
        current.next = new ListNode(l2.val);
        current = current.next;
        l2 = l2.next;
    }
    
    return head.next;
}

File: mergeTwoSortedLists/mergeTwoSortedLists.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
    head = ListNode()
    current = head
    
    while (l1 != None and l2 != None):
        if (l1.val > l2.val):
            current.next = ListNode(l2.val)
            l2 = l2.next
            
        else:
            current.next = ListNode(l1.val)                
            l1 = l1.next
            
        current = current.next               
    
    while (l1 != None):
        current.next = ListNode(l1.val)
        l1 = l1.next
        current = current.next
        
    while (l2 != None):            
        current.next = ListNode(l2.val)
        l2 = l2.next
        current = current.next
    
    return head.next